consul/datadog_checks/consul/consul.py

```diff
@@ -520,6 +449,85 @@ def check(self, _):
         if self.perform_network_latency_checks:
             self.check_network_latency(agent_dc, main_tags)
 
+    def get_service_checks(self, main_tags, nodes_per_service_tag_counts, nodes_to_service_status, service, service_tags):
+        # For every service in the cluster,
+        # Gauge the following:
+        # `consul.catalog.nodes_up` : # of Nodes registered with that service
+        # `consul.catalog.nodes_passing` : # of Nodes with service status `passing` from those registered
+        # `consul.catalog.nodes_warning` : # of Nodes with service status `warning` from those registered
+        # `consul.catalog.nodes_critical` : # of Nodes with service status `critical` from those registered
+        all_service_tags = self._get_service_tags(service, service_tags)
+        nodes_with_service = self.get_nodes_with_service(service)
```

Comment:
This line is the actual cause of the performance issue when having a large number of service., it's making an HTTP request. Instead of running all this function in a separate thread, is it possible to only make this request in the thread? Then we can process all the responses in the main thread, which is quite fast compared to the HTTP requests. E.g in the check function:

```
for service in services:
    [Make HTTP requests in a separate thread]
[Wait for threads to finish]

[Iterate over all the responses]
    [Update nodes_per_service_tag_counts, nodes_to_service_status]
    [Send nodes_* metrics]

[Continue]
```

This would be a better GIL management, and prevent this integration from slowing down other integrations too much